src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -514,32 +513,21 @@ private function getNamespacePrefixes()
      */
     public function getNamespaces()
     {
-        return (array) $this->getNamespacesObject();
-    }
+        if (!isset($this->namespaces)) {
+            $query = 'SELECT prefix, uri FROM phpcr_namespaces';
+            $result = $this->getConnection()->query($query);
+            $namespaces = $result->fetchAll(\PDO::FETCH_KEY_PAIR);
 
-    /**
-     * Return the namespaces of the current session as a referenceable ArrayObject.
-     *
-     * @return \ArrayObject
-     */
-    protected function getNamespacesObject()
-    {
-        if ($this->namespaces->count() === 0) {
-            $query = 'SELECT * FROM phpcr_namespaces';
-            $data = $this->getConnection()->fetchAll($query);
-
-            $this->namespaces->exchangeArray(array(
+            $this->namespaces = array(
                 NamespaceRegistryInterface::PREFIX_EMPTY => NamespaceRegistryInterface::NAMESPACE_EMPTY,
                 NamespaceRegistryInterface::PREFIX_JCR => NamespaceRegistryInterface::NAMESPACE_JCR,
                 NamespaceRegistryInterface::PREFIX_NT => NamespaceRegistryInterface::NAMESPACE_NT,
                 NamespaceRegistryInterface::PREFIX_MIX => NamespaceRegistryInterface::NAMESPACE_MIX,
                 NamespaceRegistryInterface::PREFIX_XML => NamespaceRegistryInterface::NAMESPACE_XML,
                 NamespaceRegistryInterface::PREFIX_SV => NamespaceRegistryInterface::NAMESPACE_SV,
-            ));
+            );
 
-            foreach ($data as $row) {
-                $this->namespaces[$row['prefix']] = $row['uri'];
-            }
+            $this->namespaces+= $namespaces;
 
             $this->originalNamespaces = $this->namespaces;
```

Comment:
good point .. maybe this is why we used \ArrayObject previously? we could of course also introduce a setter into the NodeProcessor so that we can update the namespaces.